skrub/tests/test_interpolation_join.py

```diff
@@ -76,117 +91,63 @@ def transform(self, X):
-    weather = df_module.DataFrame(weather)
-    buildings = df_module.DataFrame(buildings)
-    transformed = InterpolationJoiner(
-        weather,
-        main_key=("latitude", "longitude"),
-        aux_key=("latitude", "longitude"),
-    ).fit_transform(buildings)
-    assert ns.shape(transformed) == (2, 5)
-
-
-def test_condition_choice(df_module):
-    if df_module.name == "polars":
-        pytest.xfail(
-            reason=(
-                "In polars, DataFrame.drop() got an unexpected keyword argument 'axis'."
-            )
-        )
+def test_duplicate_column(df_module):
     main = df_module.make_dataframe({"A": [0, 1, 2]})
-    aux = df_module.make_dataframe({"A": [0, 1, 2], "rB": [2, 0, 1], "C": [10, 11, 12]})
+    aux = df_module.make_dataframe({"A": [0, 1, 2], "C": [10, 11, 12]})
     join = InterpolationJoiner(
         aux, key="A", regressor=KNeighborsRegressor(1)
     ).fit_transform(main)
     assert_array_equal(ns.to_list(ns.col(join, "C")), [10, 11, 12])
 
-    join = InterpolationJoiner(
-        aux, main_key="A", aux_key="rB", regressor=KNeighborsRegressor(1)
-    ).fit_transform(main)
-    assert_array_equal(ns.to_list(ns.col(join, "C")), [11, 12, 10])
+    # Add column with the same name twice
+    join_2 = InterpolationJoiner(
+        aux, key="A", regressor=KNeighborsRegressor(1)
+    ).fit_transform(join)
+    assert ns.shape(join_2) == (3, 3)
+    print(join_2)
+    assert re.match(r"C__skrub_[0-9a-f]+__", ns.column_names(join_2)[2])
+
+
+def test_wrong_key(df_module):
+    main = df_module.make_dataframe({"A": [0, 1, 2]})
+    aux = df_module.make_dataframe({"A": [0, 1, 2], "C": [10, 11, 12]})
 
     with pytest.raises(ValueError, match="Must pass either"):
-        join = InterpolationJoiner(
-            aux, main_key="A", regressor=KNeighborsRegressor(1)
-        ).fit(None)
+        InterpolationJoiner(aux, main_key="A", regressor=KNeighborsRegressor(1)).fit(
+            main
+        )
 
     with pytest.raises(ValueError, match="Can only pass"):
-        join = InterpolationJoiner(
+        InterpolationJoiner(
             aux, key="A", main_key="A", regressor=KNeighborsRegressor(1)
-        ).fit(None)
+        ).fit(main)
 
     with pytest.raises(ValueError, match="Can only pass"):
-        join = InterpolationJoiner(
+        InterpolationJoiner(
             aux, key="A", main_key="A", aux_key="A", regressor=KNeighborsRegressor(1)
-        ).fit(None)
+        ).fit(main)
 
 
 def test_suffix(df_module):
-    if df_module.name == "polars":
-        pytest.xfail(
-            reason=(
-                "In polars, DataFrame.drop() got an unexpected keyword argument 'axis'."
-            )
-        )
     df = df_module.make_dataframe({"A": [0, 1], "B": [0, 1]})
     join = InterpolationJoiner(
         df, key="A", suffix="_aux", regressor=KNeighborsRegressor(1)
     ).fit_transform(df)
     assert_array_equal(ns.column_names(join), ["A", "B", "B_aux"])
 
 
-def test_mismatched_indexes(df_module):
```

Comment:
why do we remove this test?

Reply:
The case where we join [0, 1] on [1, 0] is tested in `test_fit_on_none`. I can separate into two tests if necessary

Reply:
I readded a test (for pandas only). Note that because of `concat_horizontal`, the initial index is reset. We could store it and reapply it in the case of pandas

It's possible to store the index of the first arg of `concat_horizontal` and set the output to have this index, WDYT ?